google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/CopyBackupRequest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.admin.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.bigtable.admin.v2.internal.NameUtil;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.util.Timestamps;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.threeten.bp.Instant;
+
+/** Fluent wrapper for {@link com.google.bigtable.admin.v2.CopyBackupRequest} */
+public final class CopyBackupRequest {
+  private final com.google.bigtable.admin.v2.CopyBackupRequest.Builder requestBuilder =
+      com.google.bigtable.admin.v2.CopyBackupRequest.newBuilder();
+  private final String sourceBackupId;
+  private final String sourceClusterId;
+  private final String sourceInstanceId;
+  private final String sourceProjectId;
+
+  private String destClusterId;
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * instance as the destination backup. To copy from a backup in another instance, use {@link
+   * #of(String, String, String) of} method. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String sourceClusterId, String sourceBackupId) {
+    CopyBackupRequest request = new CopyBackupRequest(sourceClusterId, sourceBackupId, null, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. It assumes the source backup is located in the same
+   * project as the destination backup. To copy from a backup in another project, use {@link
+   * #of(String, String, String, String) of} method
+   */
+  public static CopyBackupRequest of(String sourceClusterId, String sourceBackupId, String sourceInstanceId) {
+    CopyBackupRequest request = new CopyBackupRequest(sourceClusterId, sourceBackupId, sourceInstanceId, null) {
+, null);
+    return request;
+  }
+
+  /**
+   * Create a {@link CopyBackupRequest} object. The source backup could be located in the same or
+   * different cluster and/or instance and/or project.
+   */
+  public static CopyBackupRequest of(
+      String sourceClusterId, String sourceBackupId, String sourceInstanceId, String sourceProjectId) {
```

Comment:
I think the method names need to be a bit more descriptive...its pretty easy to mess this up and hard to know what field mean what. Also the ordering seems a bit backwards.

Proposal:
// keep of as the bare minimum fields
CopyBackupRequest from(String sourceClusterId, String sourceBackupId)

// for a different instance
CopyBackupRequest fromAnotherInstance(String instanceId, String sourceClusterId, String sourceBackupId)

// for a different project/instance, keeping the params in hierarchical order 
CopyBackupRequest fromAnotherProject(String projectId, String instanceId, String sourceClusterId, String sourceBackupId)

Reply:
Done